prereise/gather/griddata/hifld/data_process/transmission.py

```diff
@@ -283,6 +283,30 @@ def map_via_neighbor_voltages(lines, neighbors, func, method_name):
     lines["VOLTAGE"] = lines["VOLTAGE"].astype(float)
 
 
+def create_buses(lines):
+    """Using line and substation information, create a bus within each substation and a
+    mapping of buses to substations.
+
+    :param pandas.DataFrame lines: data frame of lines.
+    :return: (*tuple*) --
+        (*pandas.Series*) -- mapping of buses to voltages.
+        (*pandas.Series*) -- mapping of buses to substations.
+    """
+    sub_1_lines = pd.Series(lines.groupby("SUB_1_ID").groups).apply(set)
+    sub_2_lines = pd.Series(lines.groupby("SUB_2_ID").groups).apply(set)
+    sub_lines = sub_1_lines.combine(sub_2_lines, set.union, fill_value=set())
+    # Get the set of voltages for any line(s) connected to each substation
+    substation_voltages = sub_lines.map(lambda x: sorted(set(lines.VOLTAGE.loc[x])))
+    buses = substation_voltages.explode()
+    buses = buses.loc[~buses.isna()]
+    buses = buses.astype(float)
+    buses.index.name = "sub_id"
+    buses = buses.to_frame(name="baseKV").reset_index()
+    bus2sub = buses.pop("sub_id")
+
+    return buses.squeeze(), bus2sub
```

Comment:
So we are returning two series without specifying `bus_id` here. Where do you envision that we are going to create a `bus` and `sub` data frame?

Reply:
I kept `buses` as just a Series for now, since we only have one column of real information that would go into the `bus` frame. We could equivalently return a data frame containing the `baseKV` and `bus2sub` columns, and then pop `bus2sub` later.

`sub` should come from the `substations` frame elsewhere in this module.

Reply:
I dimly remembered we thought about removing `bus2sub` table by adding a `sub_id` column in `bus` table given the `interconnect` column is already in `bus` table previously. But I forgot why we didn't do so. I personally lean towards returning a draft `bus` data frame with `sub_id` and `basekV` columns and then we add other columns elsewhere later to complete a `bus` data frame.

Reply:
The exact configuration of the `bus` table in PowerSimData may be a holdover from the Matpower compatibility days. Either way, since `bus` and `bus2sub` should have identical indices, we can keep them together for now and split them out later.

Reply:
Done.